x/mongo/driver/auth/sasl.go

```diff
@@ -20,6 +20,7 @@ type SaslClient interface {
 	Start() (string, []byte, error)
 	Next(challenge []byte) ([]byte, error)
 	Completed() bool
+	ShorterConversationSupported() bool
```

Comment:
Another possible way of doing this is to add an interface named something like `ShortConversationSaslClient`, having the SCRAM SASL adapter implement it, and then doing a type assertion in `ConductSaslConversation` similar to the one for `SaslClientCloser`.

I don't really have a preference for which way this is done, so I'm open to changing it if someone thinks adding another interface would be better.

Reply:
The intention for the server API is that the options are mechanism-specific.   So in Java driver implementation we added a method that allows any sasl mechanism to append whatever options it wants.  The default is to append nothing, and for SCRAM mechanisms to append this new option.